chcemvediet/apps/anonymization/cron.py

```diff
@@ -146,3 +142,92 @@ def attachment_normalization():
+        )
+    cron_logger.info(u'Recogized attachment_normalizon using mocked abbyyocr11: {}'.format(
+        attachment_normalization))
+
+def recognize_using_ocr(attachment_normalization):
+    if settings.MOCK_OCR:
+        recognize_using_mock(attachment_normalization)
+        return
+
+    try:
+        p = None
+        with temporary_directory() as directory:
+            filename = os.path.join(directory, u'file.pdf')
+            shutil.copy2(attachment_normalization.file.path, filename)
+            p = subprocess32.run(
+                [u'abbyyocr11', u'--recognitionLanguage', u'Slovak', u'--splitDualPages', u'-if',
+                 filename, u'-f', u'ODT', u'--rtfKeepLines', u'--rtfRemoveSoftHyphens',
+                 u'--rtfPageSynthesisMode', u'ExactCopy', u'-of',
+                 os.path.join(directory, u'file.odt')],
+                stdout=subprocess32.PIPE,
+                stderr=subprocess32.PIPE,
+                timeout=OCR_TIMEOUT
+            )
+            with open(os.path.join(directory, u'file.odt'), u'rb') as file_odt:
+                AttachmentRecognition.objects.create(
+                    attachment=attachment_normalization.attachment,
+                    successful=True,
+                    file=ContentFile(file_odt.read()),
+                    content_type=content_types.ODT_CONTENT_TYPE,
+                    debug=u'STDOUT:\n{}\nSTDERR:\n{}'.format(p.stdout, p.stderr)
+                )
+            cron_logger.info(u'Recognized attachment_normalization using OCR: {}'.format(
+                attachment_normalization))
+    except Exception as e:
+        trace = unicode(traceback.format_exc(), u'utf-8')
+        stdout = p.stdout if p else getattr(e, u'stdout', u'')
+        stderr = p.stderr if p else getattr(e, u'stderr', u'')
+        AttachmentRecognition.objects.create(
+            attachment=attachment_normalization.attachment,
+            successful=False,
+            content_type=content_types.ODT_CONTENT_TYPE,
+            debug=u'STDOUT:\n{}\nSTDERR:\n{}\n{}'.format(stdout, stderr, trace)
+        )
+        cron_logger.error(u'Recognizing attachment_normalization using OCR has failed: {}\n An '
+                          u'unexpected error occured: {}\n{}'.format(
+                          attachment_normalization, e.__class__.__name__, trace))
+
+def skip_recognition(attachment_normalization):
+    AttachmentRecognition.objects.create(
+        attachment=attachment_normalization.attachment,
+        successful=False,
+        content_type=None,
+        debug=u'Not supported content type'
+    )
+    cron_logger.info(u'Skipping recognition of attachment_normalization with not supported content '
+                     u'type: {}'.format(attachment_normalization.attachment))
+
+def recognize_attachment(attachment_normalization):
+    if attachment_normalization is None:
+        return
+    elif attachment_normalization.content_type == content_types.PDF_CONTENT_TYPE:
```

Comment:
Je takato struktura kodu v poriadku? Vyuzijeme niekedy vobec `skip_recognization()` a ci treba dodatcne kontrolovat `PDF_CONTENT_TYPE`? Chcel som to spravit analogicky ku funkcii `normalize_attachment`.

Reply:
Kedze uz pri selectovani `AttachmentNormalization` mame podmienku `content_type=content_types.PDF_CONTENT_TYPE`, tak `attachment_normalization.content_type == content_types.PDF_CONTENT_TYPE` bude vzdy true. Takze k vetve `skip_recognition` sa nikdy nedostaneme. Cize `skip_recognition` mozeme uplne vyhodit.

Reply:
:ok: